Sms feature

shadycuz opened the pull request:
Do not Merge, this needs attention. Fails to call send_twilio()

shadycuz commented:
Alarms are up and working. This is now safe to merge =). When should I add the Twilio Documentation to the wiki? Also what do you think about including the host the alarm is coming from in all the alerts? I have 3 running netdata and already I'm sick of looking in the URL included in the alarm notification for the name of the host. It's kinda silly if you only have 1 install though, but maybe we can find a decent solution when the multi server dashboard is built?

ktsaou commented on plugins.d/alarm-notify.sh:
```diff
@@ -568,6 +593,38 @@ EOF
+                --data-urlencode "To=${user}" \
+                --data-urlencode "Body=${message}" \
+                -u "${accountsid}:${accounttoken}" \
+                "https://api.twilio.com/2010-04-01/Accounts/${accountsid}/Messages.json")
```
You have not used the `${title}`, this is why the host is missing.

I suggest to set it first in `Body`, before the `${message}`.

shadycuz replied:
I got that code from you remember =), yes I will get that working and tested.

shadycuz replied:
@ktsaou How should I space it? so the title and msg on the same line? /n didn't work it showed up in the text msg.

ktsaou commented on plugins.d/alarm-notify.sh:
```diff
@@ -535,7 +560,7 @@ send_pushover() {
 
 send_pushbullet() {
-    local userapikey="${1}" recipients="${2}"  title="${3}" message="${4}" httpcode sent=0 user
+    local usersid="${1}" recipients="${2}"  title="${3}" message="${4}" httpcode sent=0 user
```
oops! why this needs to change?

shadycuz replied:
It doesn't =/, It's a bug in my Bash for windows 10. I had a ssh shell crash when I was editing it in vi, then when I reopened I recovered the file, but for some reason vi would randomly open the older corrupted one in swap. I changed it on accident and pushed it, fixed it and pushed it again. While changing something else bash must have opened my older corrupted file and that's how it ended back into the commit. I'll fix it in an hour after I eat.

ktsaou commented on conf.d/health_alarm_notify.conf:
```diff
@@ -133,6 +136,26 @@ SEND_PUSHBULLET="YES"
+# multiple recipients can be given like this:
+#                  "+15555555555 +17777777777"
+
+# enable/disable sending pushbullet notifications
```
not pushbullet: twilio

shadycuz replied:
You do got a point, I feel like a looked this over a 100 times last night.

ktsaou commented on plugins.d/alarm-notify.sh:
```diff
@@ -795,6 +852,17 @@ The source of this alarm is line ${src}"
+Chart: ${chart}
+Family: ${family}
+To View Netdata go to: ${goto_url}
+The source of this alarm is line ${src}"
```
SMS are normally limited to 140 characters.
Since you are posting a URL and the source line, each notification will need multiple SMS messages. This is a billing issue actually. For each notification sent, twilio will most probably charge you many SMS.

My suggestion is to not send the URL and source via twilio, to keep the message short.

shadycuz replied:
I can do that, but so far each alarm has come in as one msg. I didn't like having the URL in the msg any ways, lots of clutter and I don't expect much people to actually use netdata on the phone unless it's an emergency and they know how to reach the dashboard.

ktsaou commented:
I made a few comments. The only bug I could find is the variable you changed at `send_pushbullet()`.

I suggest to rebase your repo, to eliminate all these commits from appearing in the netdata commit log. Run this:

```sh
git rebase -i master
```

shadycuz commented:
@ktsaou I'm heading out for breakfast. I added the title before msg in body but I will want to test to see what it looks like, but was that how you thought it should be? I thought maybe adding it with the rest of the alarm message in the call to send_twilio(). I also fixed that variable I changed, and that pesky pushbullet reference in the health_alarm_contify.conf

shadycuz commented:
@ktsaou so... the commit name is really ugly lol, and it took an hour and a half... but I finally got it to rebase lol...

shadycuz commented:
@ktsaou I dare say it's ready, though I'm worried about the rebase some small change somewhere got undone, but I looked it over about 12 times and I think it's good. I have added all the small changes from your review. Took out the URL in the alert. Added in the ${title} before ${message} and tested it and it looks really good.  Now it's time to get some sleep lol

ktsaou commented:
nice! I can't find an issue either.

Could you please update the wiki before merging this?

shadycuz commented:
@ktsaou  ![image](https://cloud.githubusercontent.com/assets/17090999/20034652/620b6100-a39b-11e6-96af-4f83b8e830e2.png)

Okay, the wiki is done. I'm going to add that image and this will be ready to merge.

ktsaou commented:
merged! thanks!
